### ISSUE TITLE
hard-break-spaces false positive

### ISSUE THREAD

**Issue by dirk-thomas** [wooorm/remark-lint#31]
I have a markdown file with the following markup:

```
-   A bullet with two trailing spaces  
    Next line obviously starting with two spaces.

```

The linter shows the warning: `Use two spaces for hard line break` but that is exactly what the example does.

From looking into a code the `contents.slice(start, end)` (https://github.com/wooorm/mdast-lint/blob/50f350b02f8d0859663c45a5760f0767e9195670/lib/rules/hard-break-spaces.js#L49) seems to contain not only the two trailing spaces and the newline but also the two leading spaces of the next line. Therefore the length is greater then 3 hence the warning.

Arguably I could remove the leading spaces but I think the rule should not trigger in this case.

**Comment by wooorm** [comment]
Thanks @dirk-thomas for the issue, and the extensive description!

It’ll take a few days before I can dig-in. In the mean time, could you supply some more information? E.g., the surrounding text, and the settings you used?

**Comment by dirk-thomas** [comment]
I have updated the snippet (to use the default setting with more spaces). Then it happens without any custom configuration file.

There doesn't have to be a surrounding text. The snippet can be the only content in the file.

**Comment by wooorm** [comment]
Fixed! Had to do with the block being indented (the same would happen when inside block-quotes).

Anyway, fixed by checking just before the newline :+1:

**Comment by dirk-thomas** [comment]
Great, thank you for addressing the issue.
